activesupport/lib/active_support/test_case.rb

```diff
@@ -41,11 +41,22 @@ def test_order
       end
 
       def run(reporter, options = {})
-        if options[:patterns] && options[:patterns].any? { |p| p =~ /:\d+/ }
-          options[:filter] = \
-            Testing::CompositeFilter.new(self, options[:filter], options[:patterns])
-        end
+        line_filter = options[:patterns] && options[:patterns].any? { |p| p =~ /:\d+/ }
+        if line_filter
+          options[:patterns].each do |pattern|
+            file, *line_numbers = pattern.split(':')
 
+            if line_numbers.any?
+              line_numbers.each do |line| 
+                options[:filter] = \
+                  Testing::CompositeFilter.new(self, options[:filter], file, line) 
+              end
+            else
+              options[:filter] = \
+                Testing::CompositeFilter.new(self, options[:filter], file) 
+            end
+          end
+        end
```

Comment:
Active Support Test Case shouldn't change, all the action should be in the `CompositeFilter`